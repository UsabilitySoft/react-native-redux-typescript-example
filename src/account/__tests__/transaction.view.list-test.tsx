import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import TransactionViewList from '../transaction.view.list';
// import { initialState } from '../transaction.reducer';
const mockStore = configureStore([thunk]);

describe('Transaction List', () => {
  let store;
  let component: any;

  beforeEach(() => {
    // store = mockStore({
    //   transaction: initialState
    // });

    store = mockStore({
      transaction: {
        transactions: [
          {
            id: 1,
            currency: 'Bø',
            amount: '20.00',
            counterparty: 'Bó',
            category: 'Gifts'
          }
        ],
        transactionsLoaded: true,
        transactionsLoadedAt: 1584506435876
      }
    })

    component = renderer.create(
      <Provider store={store}>
        <TransactionViewList />
      </Provider>
    );
  });

  it('Should render with transactions state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  // We would need to simulate events, we can also test actions and reducers, skipping TDD due to time limitation
  // it('should dispatch an action on button click', () => {
  // });
});