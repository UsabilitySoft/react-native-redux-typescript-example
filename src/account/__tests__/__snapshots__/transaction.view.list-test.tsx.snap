// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction List Should render with transactions state from Redux store 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 30,
        "fontWeight": "100",
      }
    }
  >
    BóCoin
     
    Balance
  </Text>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "alignSelf": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 50,
          "fontWeight": "100",
        }
      }
    >
      20
    </Text>
    <Text
      style={
        Object {
          "fontSize": 30,
          "fontWeight": "400",
          "paddingTop": 19,
        }
      }
    >
       
      Bø
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={
        Object {
          "testUri": "../../../src/theme/assets/dummy-buttons.png",
        }
      }
      style={
        Object {
          "height": 78,
          "marginTop": 20,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={
        Object {
          "testUri": "../../../src/theme/assets/dummy-graph.png",
        }
      }
      style={
        Object {
          "height": 210,
          "width": "100%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "200",
        "paddingBottom": 20,
      }
    }
  >
    Wednesday 6 September
  </Text>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "alignSelf": "center",
        "borderColor": "#9a9a9a",
        "borderTopWidth": 0.5,
        "flexDirection": "row",
        "minHeight": 70,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#222",
            "fontSize": 20,
            "fontWeight": "300",
          }
        }
      >
        Bó
      </Text>
      <Text
        style={
          Object {
            "color": "#222",
            "fontSize": 16,
            "fontWeight": "200",
          }
        }
      >
        Gifts
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 20,
          }
        }
      >
        20.00
      </Text>
      <Text
        style={
          Object {
            "color": "#888",
            "fontSize": 16,
            "paddingLeft": 5,
            "paddingTop": 4,
          }
        }
      >
        Bø
      </Text>
    </View>
  </View>
</View>
`;
